libnetwork/datastore/datastore.go

```diff
@@ -261,6 +261,29 @@ func (ds *Store) Map(key string, kvObject KVObject) (map[string]KVObject, error)
 	return results, nil
 }
 
+// DeleteObjectIdempotent deletes a kvObject from the on-disk DB and the
+// in-memory cache. Unlike DeleteObjectAtomic, it doesn't check the optimistic
+// lock of the passed kvObject.
+func (ds *Store) DeleteObjectIdempotent(kvObject KVObject) error {
```

Comment:
Slightly wondering if "idempotent" is the correct name here. I would consider "idempotent" to be "ignore if it doesn't exist", but the description here feels more like a "force remove" (ignore if changes were made, and ignore non-existing).

Note that it's common for `--force` to ignore "not existing", so I guess that terminology would still "fit"?

With that in mind; I'm almost tempted to;

- implement a `Delete` function with a `force` option (?)
- Perhaps implement `DeleteAtomic` to be a wrapper for `Delete` (with `force == false`)

Reply:
I renamed `DeleteObjectIdempotent` to `DeleteObject`.

> implement a Delete function with a force option (?)

I'd prefer having a dedicated function than a boolean flag as it makes the semantics of each function clearer, and it's easier to spot which code path relies on the older function. We already have too much functions with frankenstein semantics, so I don't think it's worth de-duplicating < 10 loc.